src/config/cluster-config.c

```diff
@@ -60,17 +60,42 @@ int Read_Cluster(XML_NODE node, void *d1, __attribute__((unused)) void *d2) {
             } else if (strspn(node[i]->content, C_VALID) < strlen(node[i]->content)) {
                 merror("Detected a not allowed character in node name: \"%s\". Characters allowed: \"%s\".", node[i]->content, C_VALID);
                 return OS_INVALID;
+            } else if ((strcasecmp(node[i]->content, "$NODE_NAME") == 0)||(strcasecmp(node[i]->content, "$HOSTNAME") == 0)) {
+                // Get environment variables
+                char * node_name = wm_node_name();
+
+                if (node_name) {
+                    free(Config->node_name);
+                    os_strdup(node_name, Config->node_name);
+                } else {
+                    mwarn("Cannot find environment variable 'NODE_NAME'");
+                }
+            }else {
+                free(Config->node_name);
+                os_strdup(node[i]->content, Config->node_name);
             }
-            os_strdup(node[i]->content, Config->node_name);
         } else if (!strcmp(node[i]->element, node_type)) {
             if (!strlen(node[i]->content)) {
                 merror("Node type is empty in configuration");
                 return OS_INVALID;
-            } else if (strcmp(node[i]->content, "worker") && strcmp(node[i]->content, "client") && strcmp(node[i]->content, "master") )  {
-                merror("Detected a not allowed node type '%s'. Valid types are 'master' and 'worker'.", node[i]->content);
+            } else if (strcasecmp(node[i]->content, "$NODE_TYPE") == 0) {
```

Comment:
Please, let the reading of `node_type` as before, that it is, without allowing the environment variable `$NODE_TYPE`. It is not necessary for now.